pkg/apis/core/validation/shoot.go

```diff
@@ -495,21 +494,6 @@ func validateKubeControllerManagerUpdate(newConfig, oldConfig *core.KubeControll
 	return allErrs
 }
 
-func validateAddonsUpdate(new, old *core.Addons, shootUseAsSeed bool, fldPath *field.Path) field.ErrorList {
```

Comment:
I just figured out, that there is no similar validation for Shoots referenced by ManagedSeeds.

There is validation that denies enabling the ingress controller on ManagedSeeds if the referenced Shoot already enables the addon: https://github.com/gardener/gardener/blob/2e3d605831d1628db914d24837d3bd385a246dff/plugin/pkg/managedseed/validator/admission.go#L282-L285

However, there is no validation for the other direction. I.e., you can update the Shoot to enable the nginx ingress addon, even if the referencing ManagedSeed already enables the addon.


This is only partly related to this PR, but I'm wondering whether we such add such validation.
(Maybe it's not worth the effort though and rather an edge case that doesn't need validation)
Looking for opinions :)

Reply:
> However, there is no validation for the other direction. I.e., you can update the Shoot to enable the nginx ingress addon, even if the referencing ManagedSeed already enables the addon.

Valid point, it seems that I overlooked this case. For now I can try to preserve the existing logic. Instead of checking for the `shootUseAsSeed` annotation, I can try to check if the Shoot is ManagedSeed, and if yes -> forbid enabling the nginx-ingress addon from the Shoot spec.

Reply:
I decided to add this validation as part of the PR that adds the VPA "enforcement" - ref https://github.com/gardener/gardener/pull/6388 (WIP for now as I have to test thoroughly and add docs about the new validations that we introduce).

PS. Now going one more time over the discussion, I guess I also need to do the changes in `ShootValidation` admission plugin because the `ManagedSeed` admission plugin won't be invoked on changes to the Shoot spec.
TL;DR let's do this as part of https://github.com/gardener/gardener/pull/6388. As you shared in today's master we don't have such validation for ManagedSeeds (only for the legacy `use-as-seed` annotation).

Reply:
Thank you for taking care! Sounds good to me :)